fcrepo-http-api/src/test/java/org/fcrepo/integration/http/api/AbstractResourceIT.java

```diff
@@ -423,4 +456,24 @@ protected static void assertDeleted(final String id) {
         assertThat("Expected object to be deleted", getStatus(new HttpHead(location)), is(GONE.getStatusCode()));
         assertThat("Expected object to be deleted", getStatus(new HttpGet(location)), is(GONE.getStatusCode()));
     }
+
+    class RequestThread extends Thread {
```

Comment:
Rather than extending `Thread` and manually staging the executions, consider using calling `java.util.concurrent.Executors.newSingleThreadExecutor` to get an `Executor`, then calling `execute` sequentially on that with lambdas for each `Runnable`. That will guarantee the order of your two tasks being executed and should read _much_ more concisely.